test/integration/IOS/ElementTest.cs

```diff
@@ -35,20 +35,23 @@ public void FindByAccessibilityIdTest()
             By byAccessibilityId = new ByAccessibilityId("ComputeSumButton");
             Assert.Multiple(() =>
             {
-                Assert.That(Is.Not.EqualTo(_driver.FindElements(MobileBy.ClassName("UIAWindow"))[1].FindElement(byAccessibilityId).Text), null);
-                Assert.That(_driver.FindElements(MobileBy.ClassName("UIAWindow"))[1].FindElements(byAccessibilityId), Is.Not.Empty);
+                Assert.That(_driver.FindElement(MobileBy.ClassName("UIAWindow")).FindElement(byAccessibilityId).Text, Is.Not.Null);
+                Assert.That(_driver.FindElement(MobileBy.ClassName("UIAWindow")).FindElements(byAccessibilityId), Is.Not.Empty);
             });
         }
 
         [Test]
-        public void FindByByIosUiAutomationTest()
+        public void FindElementsByClassNameTest()
         {
-            By byIosUiAutomation = new ByIosUIAutomation(".elements().withName(\"Answer\")");
-            Assert.Multiple(() =>
-            {
-                Assert.That(_driver.FindElements(MobileBy.ClassName("UIAWindow"))[1].FindElement(byIosUiAutomation).Text, Is.Not.Null);
-                Assert.That(_driver.FindElements(MobileBy.ClassName("UIAWindow"))[1].FindElements(byIosUiAutomation), Is.Not.Empty);
-            });
+            By byClassName = new ByClassName("XCUIElementTypeTextField");
+            Assert.That(_driver.FindElements(byClassName), Has.Count.EqualTo(2));
```

Comment:
I usually add timers for conditions like this, e.g. "verify the amount of found elements is equal or more than 2 within 5 seconds ". They work better in slow CI env and are more reliable in general.

Reply:
We are using `_driver.Manage().Timeouts().ImplicitWait = Env.ImplicitTimeoutSec;`
isn't that enough in this case?

Reply:
The way implicit timeouts are applied only helps to wait for until any element exists. This means such wait will be finished if the given locator matches at least one element in the example above instead of two, which creates a potential race condition for apps where elements pop up on screen in serial order

Reply:
I've harnessed the WebDriverWait usage for that cause. looks better?